### PR TITLE
Expose numRetries  to configure maxRetries on error in the S3CrtClient Interface

### PR DESCRIPTION
**Pull request by joviegas**
## Motivation and Context

- Customers wanted to set NumberOfRetries for S3CrtClient

## Modifications

Directly expose StandardRetryOptions as exposed by ClientOptions, then set the MaxRetry from S3CrtClientBuilder.numRetries 
## Testing

- Junits , to test the getter.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: services/s3/src/main/java/software/amazon/awssdk/services/s3/crt/S3CrtRetryConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        return new S3CrtRetryConfiguration.DefaultBuilder(this);
+    }
+
+    public interface Builder extends CopyableBuilder<S3CrtRetryConfiguration.Builder, S3CrtRetryConfiguration> {
+
+        /**
+         * Configure the maximum number of times that a single request should be retried.
+         * @param numRetries
+         * @return The builder of the method chaining.
+         */
```
Comment (millems) [inline services/s3/src/main/java/software/amazon/awssdk/services/s3/crt/S3CrtRetryConfiguration.java:86]: Can we clarify in the docs: Is this the number of retries for a request attempt? E.g. if a putObject is broken up into 5 operations (initiate, uploadPart, uploadPart, uploadPart, complete), is this the number of retries for the putObject or the number of retries for each operation?

File: services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/DefaultS3CrtAsyncClient.java [hunk 2]
```diff
@@ -106,11 +109,16 @@ private static S3CrtAsyncHttpClient.Builder initializeS3CrtAsyncHttpClient(Defau
...
                                        .credentialsProvider(builder.credentialsProvider)
                                        .readBufferSizeInBytes(builder.readBufferSizeInBytes)
-                                       .httpConfiguration(builder.httpConfiguration)
-                                       .build();
+                                       .httpConfiguration(builder.httpConfiguration);
 
+        if (builder.retryConfiguration != null) {
+            nativeClientBuilder.standardRetryOptions(
+                new StandardRetryOptions()
+                    .withBackoffRetryOptions(new ExponentialBackoffRetryOptions()
```
Comment (millems) [inline services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/DefaultS3CrtAsyncClient.java:117]: Does this jitter the retries?

File: services/s3/src/main/java/software/amazon/awssdk/services/s3/crt/S3CrtRetryConfiguration.java (hunk 1 above)
Comment (joviegas) [reply services/s3/src/main/java/software/amazon/awssdk/services/s3/crt/S3CrtRetryConfiguration.java]: Added some more clarification that the rerties are for the Http requests/ individual operations or Service calls and not for the entire putObject
[Reference](https://awslabs.github.io/aws-crt-java/software/amazon/awssdk/crt/io/ExponentialBackoffRetryOptions.html)  `Configuration options for the exponential backoff retry strategy for http requests`

File: services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/DefaultS3CrtAsyncClient.java (hunk 2 above)
Comment (joviegas) [reply services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/DefaultS3CrtAsyncClient.java:117]: The interval between the retries are Jittered https://awslabs.github.io/aws-crt-java/software/amazon/awssdk/crt/io/ExponentialBackoffRetryOptions.JitterMode.html

File: services/s3/src/main/java/software/amazon/awssdk/services/s3/crt/S3CrtRetryConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        return new S3CrtRetryConfiguration.DefaultBuilder(this);
+    }
+
+    public interface Builder extends CopyableBuilder<S3CrtRetryConfiguration.Builder, S3CrtRetryConfiguration> {
+
+        /**
+         * Sets the maximum number of retries for a single HTTP request.
+         * <p> For example, if an upload operation is split into 4 HTTP service requests ( One for initiate,  Three for
+         * uploadPart and one for completeUpload), then numRetries specifies the maximum number of retries for each failed
+         * request, not for the entire uploadObject operation.
```
Comment (millems) [inline services/s3/src/main/java/software/amazon/awssdk/services/s3/crt/S3CrtRetryConfiguration.java:86]: s/4/5/

**Review by millems — APPROVED** [review]
One nit, otherwise LGTM

**Pull request merged**